s4e-web/src/app/utils/initializer/config.service.ts

```diff
@@ -35,14 +33,16 @@ export class S4eConfig implements IConfiguration {
     this.generalErrorKey = '__general__';
     this.maxZoom = 12;
     this.timezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+
+    this.sessionService.init(this);
   }
 
-  loadConfiguration(): Promise<IRemoteConfiguration> {
-    return combineLatest([
-      this.http.get<IRemoteConfiguration>(`api/v1/config`).pipe(tap(config => this.init(config))),
-      this.profileService.get$()
-    ]).pipe(
-      map(([configuration, profile]) => configuration)
-    ).toPromise();
+  loadConfiguration(): Promise<S4eConfig> {
+    return this.http.get<IRemoteConfiguration>(`api/v1/config`)
+      .pipe(
+        tap(config => this.init(config)),
+        switchMap(() => this.sessionService.getProfile$()),
+        map(() => this)
```

Comment:
What are you doing in this line: `map(() => this)`?

Reply:
`loadConfiguration()` is returning `S4EConfig` instance so I need to return it. I'm doing switch map earlier to load Profile information, so I need to change the context.